Src/ILGPU/Runtime/ExchangeBuffer.cs

```diff
@@ -64,6 +65,82 @@ public static class ExchangeBuffer
             var gpuBuffer = accelerator.Allocate<T>(extent);
             return new ExchangeBuffer<T>(gpuBuffer, mode);
         }
+
+        /// <summary>
+        /// Allocates a new 2D exchange buffer that allocates the specified amount of
+        /// elements on the current accelerator. Furthermore, it keeps a buffer of the
+        /// same size in pinned CPU memory to enable asynchronous memory transfers
+        /// </summary>
+        /// <typeparam name="T">The element type.</typeparam>
+        /// <param name="accelerator">The associated accelerator to use.</param>
+        /// <param name="extent">The extent (number of elements to allocate).</param>
+        /// <returns>The allocated 2D exchange buffer.</returns>
+        /// <remarks>
+        /// This function uses the default buffer allocation mode
+        /// <see cref="ExchangeBufferMode.PreferPagedLockedMemory"/>
+        /// </remarks>
+        public static ExchangeBuffer2D<T> AllocateExchangeBuffer<T>(
+            this Accelerator accelerator,
+            Index2 extent)
+            where T : unmanaged => accelerator.AllocateExchangeBuffer<T>(extent, ExchangeBufferMode.PreferPagedLockedMemory);
```

Comment:
Please ensure a maximum line length of 90 characters. There are only a few places which violate these constraints in the ILGPU compiler code base.